src/com/theironyard/novauc/Main.java

```diff
@@ -1,8 +1,134 @@
 package com.theironyard.novauc;
+import spark.ModelAndView;
+import spark.Session;
+import spark.Spark;
+import spark.template.mustache.MustacheTemplateEngine;
+
+import java.util.ArrayList;
+import java.util.HashMap;
 
 public class Main {
 
+    static HashMap<String, User> usersList = new HashMap<>(); //when the instructions asked for a String, did it mean it wanted "asdf" or actually the String type?
+
     public static void main(String[] args) {
-	// write your code here
+
+        Spark.init();
+
+        Spark.get("/",
+                ((request, response) -> {
+                    HashMap log = new HashMap<>();
+                    return new ModelAndView(log, "login.html");
+//                    else {
+//                        log.put("name", user.name);
+//                        log.put("password", user.password);
+//                        log.put("message", user.messageList);
+//                        Session session = request.session();
+//                        return new ModelAndView(log, "/messages");
+//                    }
+                }
+                ), new MustacheTemplateEngine()
+        );
+
+        Spark.get("/messages",
+                ((request, response) -> {
+
+                    Session session = request.session();
+                    String name = session.attribute("user");
+                    System.out.println(name);
+                    User user = usersList.get(name);
+
+                    HashMap log = new HashMap<>();
+                    log.put("name", user.name);
+                    log.put("password", user.password);
+                    log.put("message", user.messageList);
+            return new ModelAndView(log, "messages.html");
+        }), new MustacheTemplateEngine()
+        );
+
+
+        Spark.post("/login",((request, response) -> {
+                //I want this route to handle:
+                //query the user for their credentials
+                //if a valid user and password are entered correctly, create a session then reroute them to messages.html.
+                //if they are a returning
+                String name = request.queryParams("user");
+                String password = request.queryParams("password");
+                User user = usersList.get(name);
+                if(password.equals(user.password)) {
+                    Session session = request.session();
+                    session.attribute("user", name);
+                    response.redirect("/messages");
+                }
+            return null;
```

Comment:
This works it would be nice if there was code that threw an Exception if they input the wrong password.